libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -771,4 +771,45 @@ int lua_print(lua_State *L) {
     return 0;
 }
 
+#if (!defined(HAL_BUILD_AP_PERIPH) || defined(HAL_PERIPH_ENABLE_RANGEFINDER))
+int lua_range_finder_handle_script_msg(lua_State *L) {
+
+    const int nargs = lua_gettop(L);
+
+    // Arg 1 => self (an instance of rangefinder_backend)
+    // Arg 2 => a float distance or a RangeFinder_State user data
+    if (nargs != 2) {
+        return luaL_argerror(L, 1, "expected 1 argument");
+    }
+
+    // check_AP_RangeFinder_Backend aborts if not found. No need to check for null
+    AP_RangeFinder_Backend * ud = *check_AP_RangeFinder_Backend(L, 1);
+
+    bool result = false;
+
+    // Check to see if the first argument is the state structure.
+    const void *state_arg = luaL_testudata(L, 2, "RangeFinder_State");
+    if (state_arg != nullptr) {
+        result = ud->handle_script_msg(*static_cast<const RangeFinder::RangeFinder_State *>(state_arg));
+
+    } else {
+        // Otherwise check to see if the argument is a number.
+        int isnum;
+        const lua_Number d = lua_tonumberx(L, 2, &isnum);
+        if (!isnum)
+            return luaL_argerror(L, 2, "argument must be RangeFinder_State or number");
```

Comment:
```suggestion
        state_temp.distance_m = luaL_checknumber(L, 2);
```
You can get a float directly, the function does the error checking for you and won't return.